prow/test/integration/test/webhook_server_test.go

```diff
@@ -118,8 +126,151 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
+			logrus.WithError(err).Infof("prowjob not found")
+		} else {
+			out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
+			if err != nil {
+				logrus.WithError(err).Infof("could not clean up cluster %s", string(out))
+			}	
+		}
+		
+	})
+	
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      "webhook-server",
+	}
+	var found bool
+	var webhookDeployment appsv1.Deployment
+	err = kubeClient.Get(ctx, key, &webhookDeployment)
+	if err != nil {
+		t.Fatalf("could not list deployment %v", err)
+	}
+	found = true
+	originalProwJob := &v1.ProwJob{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ProwJob",
+			APIVersion: "prow.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"foo":               "bar",
+				"default-me":        "enabled",
+				"admission-webhook": "enabled",
+			},
+			Name:      testServerAvailability,
+			Namespace: "default",
+		},
+		Spec: v1.ProwJobSpec{
+			Cluster: "foo",
+			Refs: &v1.Refs{
+				Repo: "bar",
+			},
+			Agent:  v1.TektonAgent,
+			Type:   v1.PresubmitJob,
+			Report: true,
+		},
+		Status: v1.ProwJobStatus{
+			State: v1.PendingState,
+		},
+	}
+	webhookDeploymentCopy := webhookDeployment.DeepCopy()
+	webhookDeploymentCopy.Spec.Template.Spec.Containers[0].LivenessProbe.PeriodSeconds = 4
+	var podList corev1.PodList
+	var webhookServerPod corev1.Pod
+	err = kubeClient.List(ctx, &podList, &client.ListOptions{})
+	if err != nil {
+		t.Fatalf("could not list pods %v", err)
+	}
+	for _, pod := range podList.Items {
+		if pod.ObjectMeta.Labels["app"] == "webhook-server" {
+			webhookServerPod = pod
+			found = true
+			break
+		}
+	}
+	var wg sync.WaitGroup
+	if !found {
+		t.Fatalf("webhook-server is not running")
+	} else {
+		wg.Add(1)
+		oldName := webhookServerPod.GetName()
+		go func() {
+			defer wg.Done()
+			for {
+				exist, newPod, err := checkNewPodInCluster(oldName, kubeClient, ctx)
+				if err != nil {
+					return
+				}
+				count := 0
+				if exist && newPod.Status.Phase == "Running" {
+					return
+				}
+				//update foo label of prowjob
+				originalProwJobCopy := originalProwJob.DeepCopy()
+				originalProwJobCopy.Labels["foo"] = strconv.Itoa(count)
+				err = kubeClient.Create(ctx, originalProwJobCopy, &client.CreateOptions{})
+				if err != nil {
+					t.Errorf("prowjob could not be created. Error: %v ", err)
+					return
+				}
+				out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
+				if err != nil {
+					t.Errorf("prowjob was not properly validated. Error: %v Output: %s", err, string(out))
+					return
+				}
+				count++
+			}
+		}()
+		if err := kubeClient.Patch(ctx, webhookDeploymentCopy, client.MergeFrom(&webhookDeployment), &client.PatchOptions{}); err != nil {
+			t.Errorf("could not patch deployment")
+		}
+		time.Sleep(3 * time.Second)
```

Comment:
waitGroup is sufficient for waiting for the go routine, I don't think sleep is needed here